extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -862,15 +867,32 @@ export class AmpA4A extends AMP.BaseElement {
       });
   }
 
+  /**
+   * @visibleForTesting
+   * @return {boolean}
+   */
+  isInNoSigningExp() {
+    return (
+      getExperimentBranch(this.win, NO_SIGNING_EXP.id) ===
+      NO_SIGNING_EXP.experiment
+    );
+  }
+
   /**
    * Start streaming response into the detached document.
    * @param {!Response} response
    * @return {boolean}
    */
   streamResponse_(response) {
-    // TODO(ccordry): get size from headers equivalent to
-    // validation flow, and double check any other values
-    // that might be set.
+    // TODO(ccordry): handle fallback case and set isVerifiedAmpCreative and
+    // creativeBody appropriately.
+    this.isVerifiedAmpCreative_ = true;
+
+    const size = this.extractSize(response.headers);
+    this.creativeSize_ = size || this.creativeSize_;
+
+    // Update priority.
+    this.updateLayoutPriority(LayoutPriority.CONTENT);
```

Comment:
isn't this too early? should update priority when we know it's (claimed to be) AMP.

Reply:
Yeah you are probably right, but I haven't implemented the check for ⚡️ or fallback yet. Added a todo to make sure I update this value where appropriate.